src/components/bmfont-text.js

```diff
@@ -0,0 +1,333 @@
+var registerComponent = require('../core/component').registerComponent;
+var THREE = require('../lib/three');
+
+var createTextGeometry = require('three-bmfont-text');
+var loadBMFont = require('load-bmfont');
+var path = require('path');
+var assign = require('object-assign');
+var createSDF = require('three-bmfont-text/shaders/sdf');
+var createMSDF = require('three-bmfont-text/shaders/msdf');
+var createBasic = require('three-bmfont-text/shaders/basic');
+
+var coreShader = require('../core/shader');
+var shaders = coreShader.shaders;
+var shaderNames = coreShader.shaderNames;
+
+var alignments = ['left', 'right', 'center'];
+var anchors = ['left', 'right', 'center', 'align'];
+var baselines = ['top', 'center', 'bottom'];
+
+var DEFAULT_WIDTH = 1; // 1 matches other AFRAME default widths... 5 matches prior bmfont examples etc.
+
+// @bryik set anisotropy to 16 because I think it improves the look
+// of large amounts of text particularly when viewed from an angle.
+var MAX_ANISOTROPY = 16;
+
+var FONT_BASE_URL = 'https://cdn.aframe.io/fonts/';
+var fontMap = {
+  'default': FONT_BASE_URL + 'DejaVu-sdf.fnt',
+  'Aileron-Semibold': FONT_BASE_URL + 'Aileron-Semibold.fnt',
+  'DejaVu': FONT_BASE_URL + 'DejaVu-sdf.fnt',
+  'Exo2Bold': FONT_BASE_URL + 'Exo2Bold.fnt',
+  'Exo2SemiBold': FONT_BASE_URL + 'Exo2SemiBold.fnt',
+  'KelsonSans': FONT_BASE_URL + 'KelsonSans.fnt',
+  'Monoid': FONT_BASE_URL + 'Monoid.fnt',
+  'SourceCodePro': FONT_BASE_URL + 'SourceCodePro.fnt',
+  'mozillavr': FONT_BASE_URL + 'mozillavr.fnt'
+};
+
+var loadedFontPromises = {};
+var loadedTexturePromises = {};
+
+coreShader.registerShader('modified-SDF', {
```

Comment:
nit: could be `modified-sdf` to not be case sensitive, could add whitelines between the properties of the shader